tests/MongoDB.Driver.Tests/Specifications/server-discovery-and-monitoring/ServerDiscoveryAndMonitoringIntegrationTestRunner.cs

```diff
@@ -55,6 +60,11 @@ public void Run(JsonDrivenTestCase testCase)
         }
 
         // protected methods
+        protected override IDisposable ConfigureTestClientRelatedDisposable()
+        {
+            return new DisposableBundle(_disposables);
```

Comment:
I don't think this works. I believe that `_disposables` is empty when this method is call. It is not until LATER when configuring failpoints that anything is added to this list, and by then it is too late for this method to see them.